server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -2006,4 +2026,93 @@ public void clearPoolSourceEntitlementRefs(Iterable<String> poolIds) {
 
         return output;
     }
+
+    /**
+     * Fetches a collection of active pool IDs mapped to a given owner. If no such pool
+     * are present, an empty collection will be returned.
+     *
+     * @param ownerId
+     *  The ID of the owner for which to fetch active Pool IDs
+     *
+     * @return
+     *  A collection of active Pool IDs belonging to the given owner.
+     */
+    public Collection<String> getActivePoolByOwnerId(String ownerId) {
+        Date currentDate = new Date();
+        String jpql = "SELECT p.id FROM Pool p WHERE p.owner.id = " +
+            " :owner_id AND p.endDate >= :endDate AND p.startDate <= :startDate";
+
+        List<String> uuids = this.getEntityManager()
+            .createQuery(jpql, String.class)
+            .setParameter("owner_id", ownerId)
+            .setParameter("endDate", currentDate)
+            .setParameter("startDate", currentDate)
+            .getResultList();
+
+        return uuids != null ? uuids : Collections.<String>emptyList();
+    }
+
+    /**
+     * Fetches a mapping of pool IDs to sets of product UUID & derived product UUID of
+     * the given pool. The returned map will only contain mappings for pools specified in the given
+     * collection of pool IDs.
+     *
+     * @param poolIds
+     *  A collection of pool IDs for which to fetch product & derived product UUIDs.
+     *
+     * @return
+     *  A mapping of pool IDs to product & derived product UUIDs
+     */
+    public Map<String, Set<String>> getProductAndDerivedProductUuidsByPoolIds(Collection<String> poolIds) {
```

Comment:
Let's try to find a better way to write this. This method looks heavily derived from the bulkSQL[update|delete] methods from AbstractHibernateCurator which should be flagged as deprecated if they're not already. Building raw SQL can be dangerous, and runs into potential problems with parameter limits (as this implementation does here).

Reply:
Noted. Changed the logic here.